### PR TITLE
Adds project sync connector

### PR DESCRIPTION
**Pull request by Alihassanc5**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        :return str: The name of the cursor field.
+        """
+        return []
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Override to determine the latest state after reading the latest record. This typically compared the cursor_field from the latest record and
+        the current state and picks the 'most' recent cursor. This is how a stream's state is determined. Required for incremental.
+        """
+        return {}
```
Comment (ehmadzubair) [inline airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py:106]: Why is this here?

File: airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                'Prosjektgruppe': row[5],
+                'Prosjektleder': row[6],
+                'Fra': row[7],
+                'Til': row[8],
+                'Kontraktsbeløp': row[9],
+                'Kostnad totalt': row[10],
+                'Tot. est. timer': row[11],
+                'Estimert dekningsgrad': row[12],
+                'Koststed': row[13]
+            }
```
Comment (ehmadzubair) [inline airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py:148]: this can be generic and moved to the superclass

File: airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    `class ProjectSyncStream(HttpStream, ABC)` which is the current class
+    `class Customers(ProjectSyncStream)` contains behavior to pull data for customers using v1/customers
+    `class Employees(ProjectSyncStream)` contains behavior to pull data for employees using v1/employees
+
+    If some streams implement incremental sync, it is typical to create another class
+    `class IncrementalProjectSyncStream((ProjectSyncStream), ABC)` then have concrete stream implementations extend it. An example
+    is provided below.
+
+    See the reference docs for the full list of configurable options.
+    """
```
Comment (ehmadzubair) [inline airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py:42]: why is this comment here?

File: airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+    # TODO: Fill in the url base. Required.
+    url_base = "https://example-api.com/v1/"
```
Comment (ehmadzubair) [inline airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py:45]: why not set base URL?

File: airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+                'Kontraktsbeløp': row[9],
+                'Kostnad totalt': row[10],
+                'Tot. est. timer': row[11],
+                'Estimert dekningsgrad': row[12],
+                'Koststed': row[13]
+            }
+
+
+class StaffTrackStream(HttpStream):
+    url_base = "https://www.xledger.net/Flex/"
```
Comment (ehmadzubair) [inline airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py:152]: This should be in the superclass

File: airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py (hunk 1 above)
Comment (Alihassanc5) [reply airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py:106]: It's auto-generated code, which can be used in the future to implement Incremental streams (if any).

File: airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py (hunk 2 above)
Comment (Alihassanc5) [reply airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py]: Fixed it!

File: airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py (hunk 3 above)
Comment (Alihassanc5) [reply airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py]: Fixed it.

File: airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py (hunk 4 above)
Comment (Alihassanc5) [reply airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py]: fixed it!

File: airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py (hunk 5 above)
Comment (Alihassanc5) [reply airbyte-integrations/connectors/source-project-sync/source_project_sync/streams.py]: fixed it!

**Comment by marcosmarxm** [comment]
@Alihassanc5 can you update the description WHAT and HOW explaining your pull request. Would be better to create first an issue.
